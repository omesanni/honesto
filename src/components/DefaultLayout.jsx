import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const DefaultLayout = props => (
  <div>
    <header className={'header'}>
      <nav className={'navbar navbar-expand-md p-0'}>
        <strong className={'navbar-brand mb-0'}>
          {'Honesto'}
        </strong>

        <div className={'collapse navbar-collapse'} id={'navbarCollapse'}>
          <ul className={'navbar-nav'}>
            <li className={'nav-item active'}>
              <Link className={'nav-link'} to={'/'}>
                {'Share Feedback'}
              </Link>
            </li>
            <li className={'nav-item'}>
              <a className={'nav-link'}>{'My Feedback'}</a>
            </li>
            <li className={'nav-item'}>
              <a className={'nav-link'}>{'Team Feedback'}</a>
            </li>
            <li className={'nav-item'}>
              <a className={'nav-link'}>{'Teams'}</a>
            </li>
          </ul>
        </div>
      </nav>
    </header>

    <div className={'container'}>
      {props.children}
    </div>
  </div>
);

DefaultLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default DefaultLayout;
