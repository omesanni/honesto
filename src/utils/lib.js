import React from 'react';
import AsyncComponent from '../components/AsyncComponent';

/**
 * Lazy loads a component
 * @param  {Function} getComponent Returns a dynamic import of the component
 * @param  {Object}   props
 * @return {ReactElement}
 */
export function lazyLoadComponent(getComponent, props = {}) {
  return React.createElement(AsyncComponent, { ...props, render: getComponent });
}
