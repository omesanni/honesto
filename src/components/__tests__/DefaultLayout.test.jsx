import React from 'react';
import { expectToMatchSnapshot } from '../../utils/tests';
import DefaultLayout from '../DefaultLayout';

it('renders', () => {
  expectToMatchSnapshot(<DefaultLayout><div /></DefaultLayout>);
});
