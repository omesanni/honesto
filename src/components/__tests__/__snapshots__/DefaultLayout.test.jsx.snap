// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <header
    className="header"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <span
        className="navbar-brand mb-0 h1"
      >
        Giphies
      </span>
    </nav>
  </header>
  <div
    className="container"
  >
    <div />
  </div>
</div>
`;
